File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program110.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user reverse that string in place.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCount;

	public:

		Solution(string str,int iCnt=0,int iCount=0)
		{
			this->str=str;
		}

		void strRevX()
		{
			iCnt=(str.length());

			while(iCnt!=-1)
			{
			
				cout<<str[iCnt];
				iCnt--;	
			}	
			cout<<"\n";
		}
};

int main()
{
	string str;

	cout<<"Enter string: ";
	getline(cin,str);
	
	Solution *sObj = new Solution(str);
	sObj->strRevX();

	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: abcd
dcba
`
*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program110.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user reverse that string in place.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Solution
{
	String str;
	int iCnt;

	Solution(String str)
	{
		this.str=str;
		this.iCnt=0;
	}

	void StrRevX()
	{
		
		iCnt=str.length()-1;
		while(iCnt!=-1)
		{
			System.out.print(str.charAt(iCnt));
			iCnt--;
		}
		
			System.out.println();
	}
}

class Program
{
	public static void main(String args[])
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		Solution obj = new Solution(str);
		obj.StrRevX();
	}
}

/*
OUTPUT:

Enter the string:
abcd
dcba


*/
